northstar/src/runtime/mount.rs

```diff
@@ -63,101 +63,271 @@ pub enum Error {
     MissingKey(String),
 }
 
-pub(super) async fn mount_npk_repository(
-    config: &Config,
-    repo: &Repository,
-) -> Result<Vec<Container>, Error> {
-    info!("Mounting NPKs from {}", repo.dir.display());
-    let mut dir = fs::read_dir(&repo.dir)
-        .await
-        .map_err(|e| Error::Io("Failed to read repository".to_string(), e))?;
-    let mut containers = Vec::new();
-    while let Ok(Some(entry)) = dir.next_entry().await {
-        containers.push(mount_npk(&config, &entry.path(), repo).await?);
-    }
+#[derive(Debug)]
+pub struct MountControl {
+    dm: dm::Dm,
+    lc: LoopControl,
+    run_dir: PathBuf,
```

Comment:
I'd keep the `run_dir` detail out of `MountControl` it's to closely coupled to the `State` impl. `MountControl` should be something generic for mounting. The pathes can be passed as arguments.